CryptKeeperZK/crypt-keeper-extension
# feat: enable backup

## Explanation

This PR enables backup feature.

## More Information

Blocked by #655
Closes #327
Closes #32 

## Screenshots/Screencaps

![image](https://github.com/CryptKeeperZK/crypt-keeper-extension/assets/14254374/dfafd21d-a283-4983-90db-db9ec65b7bc8)

![image](https://github.com/CryptKeeperZK/crypt-keeper-extension/assets/14254374/0b6ec53a-ae3e-46a5-bbb6-68ef892cfa41)

![image](https://github.com/CryptKeeperZK/crypt-keeper-extension/assets/14254374/4559def5-cdc4-4a9f-88ab-ab6288d36909)

## Manual Testing Steps

1. Go to backup and download backup data
2. Reinstall the extension
3. Upload backup
4. Check the accounts, approvals, identities

There are more advanced test cases but they will be added as e2e tests:
1. Override with user mnemonic
2. Download/upload same identity but with different metadata
3. Double upload
4. Serial download/upload
5. Empty download/upload
6. Download/Use/Upload 

## Pre-Merge Checklist


> PR template source from [github.com/MetaMask](https://github.com/MetaMask)
